File: TypeScript/Codewars/src/resolutions/Bouncing-Balls.ts
export function bouncingBall(h: number, bounce: number, window: number): number {
	let heightValue = h;

	if (heightValue <= 0 || bounce <= 0 || bounce >= 1 || window >= heightValue) {
		return -1;
	}

	let totalPasses = 0;

	while (heightValue > window) {
		totalPasses++;

		if (heightValue * bounce > window) {
			totalPasses++;
		}

		heightValue *= bounce;
	}

	return totalPasses;
}

File: Csharp/CodeWars/src/Completed/BouncingBalls.cs
namespace CodeWars.Completed;

public class BouncingBalls
{
    public static int BouncingBall(double h, double bounce, double window)
    {
        double heightValue = h;

        if (heightValue <= 0 || bounce <= 0 || bounce >= 1 || window >= heightValue)
            return -1;

        int totalPasses = 0;

        while (heightValue > window)
        {
            totalPasses++;

            if (heightValue * bounce > window)
                totalPasses++;

            heightValue *= bounce;
        }

        return totalPasses;
    }
}
